packages/shared/providers.yaml

```diff
@@ -538,6 +538,24 @@ aws:
             example: com
             pattern: '^[a-z.]+$'
 
+aws-iam:
+    display_name: AWS IAM
+    categories:
+        - dev-tools
+    auth_mode: BASIC
+    proxy:
+        base_url: https://iam.amazonaws.com/${connectionConfig.region}
```

Comment:
@bodinsamuel, I've added a `connectionConfig` region here to prompt the user to input the region. This is required for signing the requests but not for making the actual call. Setting `baseUrlOverride` at the script level isn't the most efficient approach. Another option could be to set `aws-iam` as unauthenticated, then save the username, password, and region to `connectionConfig`. However, this would mean storing credentials in an unencrypted format. Do you have any suggestions for a workaround here?

Reply:
Yes, that works, thank you Samuel will push an update for this 👍.

Reply:
I find this confusing to have:
```
proxy:
    base_url: string
    connection_base_url: string
```

I'm also unclear how this solves the issue? Can you explain further @bodinsamuel ?

Reply:
My understanding is that we only need this url `https://iam.amazonaws.com/${connectionConfig.region}` for connecting, so if we put that by default every call to aws will need to use baseUrlOverride. And if we don't do it we need a IF in our code.

If we dissociate the base_url for regular calls and token calls it could help solve that with zero specific IF in the code and good DX.

Reply:
we have `authorization_url` for oauth2 maybe a similar wording could help, or authentication since you are adding `authentication_header`

Reply:
Yes, the wording looks great 👌. Perhaps we could include it in the proxy object, as it's only needed for proxy calls, like this;

```
aws-iam:
    display_name: AWS IAM
    categories:
        - dev-tools
    auth_mode: BASIC
    proxy:
        base_url: https://iam.amazonaws.com
        authentication_header: ${connectionConfig.region}
        retry:
            at: 'x-ratelimit-reset'
    docs: https://docs.nango.dev/integrations/all/aws

Reply:
I meant you are adding `authentication_header` in your other PR, not sure this one is a header

Reply:
I think we may be misunderstanding each other because of the wording 😅, so here, I will revert to using `connection_base_url`  to dissociate base_url for regular calls, and `connection_base_url` to prompt the user to input the region.